From 4cc827d53d774ab98954a871967700dc2e5cd430 Mon Sep 17 00:00:00 2001
From: Mikhail Krichanov <krichanov@ispras.ru>
Date: Wed, 14 Jul 2021 18:49:59 +0300
Subject: [PATCH] Tried to fix MSFT Shell.

---
 MdeModulePkg/Library/UefiHiiLib/HiiLib.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/MdeModulePkg/Library/UefiHiiLib/HiiLib.c b/MdeModulePkg/Library/UefiHiiLib/HiiLib.c
index 5760282..1514afd 100644
--- a/MdeModulePkg/Library/UefiHiiLib/HiiLib.c
+++ b/MdeModulePkg/Library/UefiHiiLib/HiiLib.c
@@ -157,7 +157,9 @@ HiiAddPackages (
   // Calculate the length of all the packages in the variable argument list
   //
   for (Length = 0, VA_START (Args, DeviceHandle); (Package = VA_ARG (Args, UINT32 *)) != NULL; ) {
-    Length += (ReadUnaligned32 (Package) - sizeof (UINT32));
+    UINT32 Tmp = ReadUnaligned32 (Package);
+    ASSERT (Tmp >= sizeof (UINT32));
+    Length += (Tmp - sizeof (UINT32));
   }
   VA_END (Args);
 
-- 
2.20.1

